### ISSUE TITLE
Version 2.0.3: checkAsync runs its callbacks multiple times and worse, when the input is invalid, the pass callback is run

### ISSUE THREAD

**Issue by DanFMoore** [mikeerickson/validatorjs#73]
``` javascript
var Validator = require('validatorjs');

var validator = new Validator(
    {
        name: 'Valid name',
        email: 'not-an-email-address',
        confirm_email: 'also-invalid'
    },
    {
        name: 'required',
        email: 'required|email',
        confirm_email: 'required|email'
    }
);

validator.checkAsync(
    function() {
        console.log('passes');
    },
    function() {
        console.log('fails');
        console.log(validator.errors.all());
    }
);
```

The output is:

passes
fails
{ email: [ 'The email format is invalid.' ] }
fails
{ email: [ 'The email format is invalid.' ],
  confirm_email: [ 'The confirm email format is invalid.' ] }

When it should be:

fails
{ email: [ 'The email format is invalid.' ],
  confirm_email: [ 'The confirm email format is invalid.' ] }

When the output is valid (just change the above example to have valid email addresses), the output is:

passes
passes
passes

**Comment by DanFMoore** [comment]
This makes constructing a promise from the result almost impossible.

**Comment by DanFMoore** [comment]
I have just checked and it works as intended on 2.0.2

**Comment by garygreen** [comment]
I've had to revert to an older version temporarily which introduces a few old bugs but at least async works as it should. Will try and get round to fixing it asap @skaterdav85 would be good if you could lend a hand.

**Comment by garygreen** [comment]
Right. Should be fixed now I've just used the old AsyncResolvers thing I wrote. Released 2.0.5
